src/http/HttpRequest.ts

```diff
@@ -39,15 +50,51 @@ export class HttpRequest implements types.HttpRequest {
                 body = init.body.string;
             }
 
-            this.#uReq = new uRequest(url, {
+            uReq = new uRequest(url, {
                 body,
                 method: nonNullProp(init, 'method'),
                 headers: fromNullableMapping(init.nullableHeaders, init.headers),
             });
         }
 
-        this.query = new URLSearchParams(fromNullableMapping(init.nullableQuery, init.query));
-        this.params = fromNullableMapping(init.nullableParams, init.params);
+        const query = new URLSearchParams(fromNullableMapping(init.nullableQuery, init.query));
+        const params = fromNullableMapping(init.nullableParams, init.params);
+
+        return [uReq, query, params];
+    }
+
+    #initStreamRequest(init: InternalHttpRequestInit): [uRequest, URLSearchParams, HttpRequestParams] {
+        const proxyReq = nonNullProp(init, 'proxyRequest');
+
+        const hostHeaderName = 'x-forwarded-host';
+        const protoHeaderName = 'x-forwarded-proto';
+        const host = proxyReq.headers[hostHeaderName];
+        const proto = proxyReq.headers[protoHeaderName];
+        if (typeof host !== 'string' || typeof proto !== 'string') {
+            throw new AzFuncSystemError(`Expected headers "${hostHeaderName}" and "${protoHeaderName}" to be set.`);
+        }
+        const url = `${proto}://${host}${nonNullProp(proxyReq, 'url')}`;
+
+        let uReq = init.undiciRequest;
+        if (!uReq) {

```

Comment:
Similar idea to below. We can abstract away the code to set uReq and just have `const uReq =  constructUndiciReq(init.undiciReques);`.

This way `#initStreamRequest` itself consists of merely 5 or 6 lines and we can clearly see its key components in a glance.

Reply:
see other comment